src/pages/index.js

```diff
@@ -6,78 +7,224 @@ import PopupWithImage from "../components/PopupWithImage.js";
+      cardSection.addItem(cardElement);
+      addCardForm.close();
+      addCardForm.reset();
+      addCardValidator._disableButton();
+      addCardForm.close();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      addCardForm.renderLoading(false);
+    });
+}
+
+const addCardForm = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+addCardForm.setEventListeners();
+
+// EVENT LISTENER FOR NEW CARD BUTTON
+constants.addCardButton.addEventListener("click", () => {
+  addCardForm.open();
+  addCardValidator._disableButton();
+});
+
+// PREVIEW IMAGE MODAL
 const cardPreview = new PopupWithImage("#preview-image-modal");
 cardPreview.setEventListeners();
 
 function handleImageClick(name, link) {
   cardPreview.open({ name, link });
 }
 
-function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  return cardElement.getView();
+const deleteCardPopup = new PopupWithForm("#delete-card-modal", () => {
+  api
+    .deleteRequest(currentCardID)
+    .then(() => {
+      currentCardElement.remove();
+      deleteCardPopup.close();
+    })
+    .catch((error) => console.error("Error deleting card:", error));
+});
+
+let currentCardID = null;
+let currentCardElement = null;
+
+function handleDeleteCard(cardID, cardElement) {
+  currentCardID = cardID;
+  currentCardElement = cardElement;
+  deleteCardPopup.open();
 }
 
-// Initialize user information
-const userInformation = new UserInfo({
-  name: ".profile__title",
-  description: ".profile__description",
-});
+//-----HANDLE LIKES----/
+
+function handleLikeButton(likeButton, likedStatus, cardID) {
+  if (likedStatus) {
+    return api
+      .removeLike(cardID)
+      .then(() => {
+        likeButton.classList.remove("card__like-button_active");
+        return false;
+      })
+      .catch((error) => console.error("Error removing like:", error));
+  } else {
+    return api
+      .addLike(cardID)
+      .then(() => {
+        likeButton.classList.add("card__like-button_active");
+        return true;
+      })
+      .catch((error) => console.error("Error adding like:", error));
+  }
+}
+//----------USER INFO------//
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api
+  .getUser()
+  .then((inputValues) => {
+    userInfo.setUserInfo(inputValues.name, inputValues.about);
+    userInfo.setAvatar(inputValues.avatar);
+  })
+  .catch((err) => console.error("Error fetching user data:", err));
+
+function handleProfileEditSubmit(inputValues) {
+  profileEditForm.renderLoading(true);
+  api
+    .editProfile(inputValues.name, inputValues.about)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      profileEditForm.close();
+    })
+    .catch((err) => console.error("Error editing profile:", err))
+    .finally(() => {
+      profileEditForm.renderLoading(false);
```

Comment:
**Could be improved**


If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}

```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }

  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```